File: Sorting/ShellSort/shellSort.py
def shellSort(arr,n):
    interval = n//2
    while interval > 0:
        for i in range(interval,n):
            temp = arr[i]
            j = i
            while j>= interval and arr[j-interval] > temp :
                arr[j] = arr[j-interval]
                j -= interval
                print(arr)
            
            arr[j] = temp
        interval //=2


if __name__ == "__main__":
    data = [3,535,6,35,546,47,567,63,452,3,5,64,75,686,78,646,345,24]
    size = len(data)
    shellSort(data,size)
    print(data)

File: Sorting/ShellSort/shellSort.cpp
#include<iostream>
using namespace std;

void shellsort(int arr[],int n)
{
    for(int interval = n/2; interval>0 ; interval /= 2)
    {
        for(int i=interval ; i < n ; i++)
        {
            int temp = arr[i];
            int j;
            for(j=i; j>= interval && arr[j-interval] > temp; j -= interval)
            {
                arr[j] = arr[j-interval];
            }
            arr[j] = temp;
        }
    }
}

void printArr(int arr[],int n)
{
    for(int i=0;i<n;i++)
    {
        cout<<arr[i]<<"\t";
    }
    cout<<endl;
}

int main()
{
    int data[] = {4,7,2,6,2,7,72,8,1};
    int size = sizeof(data)/sizeof(data[0]);
    shellsort(data,size);
    cout<<"\nSorted Array\n";
    printArr(data,size);

    return 0;
}